File: typescript/agentkit/src/action-providers/aave/schemas.ts
import { z } from "zod";

/**
 * Input schema for supply action.
 */
export const SupplySchema = z
  .object({
    chain: z.string().describe("The chain to supply on (e.g. base-mainnet)"),
    amount: z.string().describe("The amount to supply"),
    asset: z.string().describe("The asset to supply (e.g. ETH, USDC)"),
  })
  .strip()
  .describe("Instructions for supplying assets to Aave");

/**
 * Input schema for withdraw action.
 */
export const WithdrawSchema = z
  .object({
    chain: z.string().describe("The chain to withdraw from (e.g. base-mainnet)"),
    amount: z.string().describe("The amount to withdraw"),
    asset: z.string().describe("The asset to withdraw (e.g. ETH, USDC)"),
  })
  .strip()
  .describe("Instructions for withdrawing assets from Aave");

File: python/coinbase-agentkit/coinbase_agentkit/action_providers/compound/schemas.py
"""Schemas for Compound action provider."""

from typing import Literal

from pydantic import BaseModel, Field


class CompoundSupplyInput(BaseModel):
    """Input schema for supplying assets to Compound."""

    asset_id: Literal["weth", "cbeth", "cbbtc", "wsteth", "usdc"] = Field(
        ...,
        description="The asset ID to supply to the Compound market, one of `weth`, `cbeth`, `cbbtc`, `wsteth`, or `usdc`",
    )
    amount: str = Field(
        ...,
        description="The amount of the asset to supply to the Compound market, e.g. 0.125 weth; 19.99 usdc",
    )


class CompoundWithdrawInput(BaseModel):
    """Input schema for withdrawing assets from Compound."""

    asset_id: Literal["weth", "cbeth", "cbbtc", "wsteth", "usdc"] = Field(
        ...,
        description="The asset ID to withdraw from the Compound market, one of `weth`, `cbeth`, `cbbtc`, `wsteth`, or `usdc`",
    )
    amount: str = Field(
        ...,
        description="The amount of the asset to withdraw from the Compound market, e.g. 0.125 weth; 19.99 usdc",
    )


class CompoundBorrowInput(BaseModel):
    """Input schema for borrowing assets from Compound."""

    asset_id: Literal["weth", "usdc"] = Field(
        ...,
        description="The asset ID to borrow from the Compound market, either `weth` or `usdc`",
    )
    amount: str = Field(
        ...,
        description="The amount of the asset to borrow from the Compound market, e.g. 0.125 weth; 19.99 usdc",
    )


class CompoundRepayInput(BaseModel):
    """Input schema for repaying borrowed assets to Compound."""

    asset_id: Literal["weth", "usdc"] = Field(
        ...,
        description="The asset ID to repay to the Compound market, either `weth` or `usdc`",
    )
    amount: str = Field(
        ...,
        description="The amount of the asset to repay to the Compound market, e.g. 0.125 weth; 19.99 usdc",
    )


class CompoundPortfolioInput(BaseModel):
    """Input schema for getting portfolio details from Compound."""

    pass  # No inputs required, matches CDP Agentkit Core implementation
